schemas/input/csl-data.json

```diff
@@ -437,6 +437,78 @@
     "additionalProperties": false
   },
   "definitions": {
+    "edtf-string": {
+      "title": "EDTF date string",
+      "description": "CSL input supports EDTF, levels 0 and 1.",
+      "type": "string",
+      "pattern": "^[0-9-%~X?./]{4,}$"
+    },
+    "date-object": {
+      "title": "Date Object",
+      "description": "An EDTF date represented as an object.",
+      "type": "object",
+      "properties": {
+        "date-parts": {
+          "type": "array",
+          "items": {
+            "type": "integer"
+          },
+          "minItems": 1,
+          "maxItems": 3
+        },
+        "literal": {
+          "title": "Literal date",
+          "description": "A date that should be passed through to the processor as is; for dates that cannot be represented in EDTF.",
+          "examples": ["Han Dynasty"],
+          "type": "string"
+        },
+        "approximate": {
+          "title": "Approximate",
+          "description": "Indicates an approximate or circa date, which may be presented as `circa May 23, 1972`.",
+          "type": "boolean"
+        },
+        "uncertain": {
+          "title": "Uncertain",
+          "description": "Indicates uncertainty about a date representation, which may be presented as `May 23, 1972?`",
+          "type": "boolean"
+        }
+      },
+      "required": ["date-parts"]
+    },
+    "date-range": {
+      "title": "Date Range",
+      "description": "An EDTF range is a two-item array. An open end or begin of a range can be represented with a null.",
```

Comment:
The null option for open currently isn't implemented. Not sure how to do this.

Reply:
Would this or permitting an empty `date-parts` array be better?

Reply:
IDK; if I don't hear from any developers, am just going to go with this and figure we can correct it before merging to master.

But I was looking for the most minimal representation of the open item. I also considered this:

``` json
 {
    "id": "four",
    "type": "book",
    "title": "The Title",
    "issued": [
      { "date-parts": [2000] },
      { "open": "true" }
    ]
  }
```